src/charm.py

```diff
@@ -173,12 +215,81 @@ def __init__(self, *args):
 
         self._set_status()
 
+    def _on_cos_agent_relation_joined(self, _):
+        self._stored.have_gagent = True
+        self._set_status()
+
+    def _on_cos_agent_relation_broken(self, _):
+        self._stored.have_gagent = False
+        self._set_status()
+
+    def _delete_existing_dashboard_files(self, dashboards_dir: str):
+        directory = Path(dashboards_dir)
+        if directory.exists():
+            for file_path in directory.glob("request_*.json"):
+                file_path.unlink()
+
+    def _create_dashboard_files(self, dashboards_dir: str):
+        dashboards_rel = self._dashboard_aggregator._target_relation
+
+        directory = Path(dashboards_dir)
+        directory.mkdir(parents=True, exist_ok=True)
+
+        for relation in self.model.relations[dashboards_rel]:
+            for k in relation.data[self.unit].keys():
+                if k.startswith("request_"):
+                    dashboard = json.loads(relation.data[self.unit][k])["dashboard"]  # type: ignore
```

Comment:
Should we assume we're getting a valid json? Anything could happen that would cause `JSONDecodeError`?